modules/saml/src/Auth/Source/SP.php

```diff
@@ -159,14 +159,12 @@ public function getHostedMetadata(): array
 
         // add NameIDPolicy
         if ($this->metadata->hasValue('NameIDPolicy')) {
-            $format = $this->metadata->getValue('NameIDPolicy');
-            if (is_array($format)) {
+            $format = $this->metadata->getArray('NameIDPolicy');
+            if ($format !== []) {
                 $metadata['NameIDFormat'] = Configuration::loadFromArray($format)->getOptionalString(
```

Comment:
I think we can simplify this by doing `$metadata['NameIDFormat'] = $format['Format'] ?? Constants::NAMEID_TRANSIENT;`